### ISSUE TITLE
File System Synchronization Fails on Android

### ISSUE THREAD

**Issue by WijMal** [laurent22/joplin#4122]
## Environment

Joplin version: 1.3.13
Platform: Android
OS specifics: 10

## Steps to reproduce

1. Configuration > Synchronization target : file system
2. Directory(absolute path) : /storage/emulated/0/Sync
     ^ the directory exists and Syncthing has no problem with read/write to it
3. Save > prompts "Allow Joplin to access photos and media on your device"
4. Allowed
5. Synchronize > errors!
    Error: Error: Directory could not be created. Path: /storage/emulated/0/Sync/locks
6. Manually created "locks" folder
    Error: Error: Directory could not be created. Path: /storage/emulated/0/Sync/temp
7. Manually created "temp" folder
    Error: Error: Could not read directory: /storage/emulated/0/Sync/locks: Attempt to get length of null array. Path: /storage/emulated/0/Sync/locks

## Describe what you expected to happen
Seems like Joplin fails to request needed "File access" android permissions at step 3.

Bug persists as of joplin v1.4.6 pre-release version

joplin v1.2.5 release works fine without running into this. It requests permissions as "Allow Joplin to access photos, media and **files** on your device" at Step 3

Joplin no longer requesting android for "File" access permissions seems to be the case

**Comment by laurent22** [comment]
At which point did the bug happened. Was it when setting up a new device?

**Comment by WijMal** [comment]
Yes, This is a fresh install on Android 10. As the 2 latest joplin android versions run into it, I hv installed v1.2.5 and it works fine.

It seems this bug was introduced sometime after v1.2.5

**Comment by gxylgbx** [comment]
Me too，joplin v1.2.6 release works fine.

**Comment by fusterjj** [comment]
This happens on Android 11 as well, although the error is slightly different: "Error: Operation not permitted. Path: /storage/emulated/0/<mydir>/locks/exclusive_mobile_xxxxxxxxxxx.json.

If I look at Joplin in the Android Settings page, it shows "Files and media" permission as "Allow access to media only".  There is no option to pick anything else other than "Deny".

I am also using Syncthing in this directory.  Joplin v1.3.13.

**Comment by nettok** [comment]
I am getting the same error in Android 10, even though I did grant the storage permission.

**Comment by ronsmits** [comment]
Can we re-open this? I just installed joplin and syncthing on a new phone (oneplus nord, android 11) and I have the same issue. Joplin version is 2.6.3 from 23 november 2021. Please advise on how to solve this

**Comment by Junelicious** [comment]
After months of use and Android 11 upgrade I am now too, stuck with that issue. Same use case as when issue was opened,  Joplin with Syncthing.

**Comment by mgsarris** [comment]
I'm having the same issue on Andorid 12 (Pixel 4a) w/ Joplin 2.6.3. Only permission Joplin seems to request for File Sync is for media.

**Comment by jwalsoe** [comment]
Same problem occurring on my Android One Device, android 11. Was also wondering why only permission to media is requested, not files as well. Joplin 2.6.3 Android, db version 41

**Comment by romelsalwi** [comment]
Hi. This issue still exists!
I just check on the permission granted to the Joplin app on Android 12 and it shows only read permissions. As I wqs trying to find a way to give it write permissions but I couldn't find a way.

**Comment by roman-r-m** [comment]
> Hi. This issue still exists! ( v2.7.2) I just check on the permission granted to the Joplin app on Android 12 and it shows only read permissions. As I was trying to find a way to give it write permissions but I couldn't find a way.

See https://github.com/laurent22/joplin/issues/5779 for the workaround

**Comment by romelsalwi** [comment]
> > Hi. This issue still exists! ( v2.7.2) I just check on the permission granted to the Joplin app on Android 12 and it shows only read permissions. As I was trying to find a way to give it write permissions but I couldn't find a way.
> 
> See #5779 for the workaround

TL DR?

**Comment by lum7671** [comment]
Galaxy Note 20 Ultra / Android 12 / Joplin 2.7.2

<img src="https://user-images.githubusercontent.com/1220430/161889418-6e43b0f1-4c4e-48f3-a7bf-19d44f5b4e71.png" width="200" />

**Comment by shuvojitk7** [comment]
> Galaxy Note 20 Ultra / Android 12 / Joplin 2.7.2
> 
> <img src="https://user-images.githubusercontent.com/1220430/161889418-6e43b0f1-4c4e-48f3-a7bf-19d44f5b4e71.png" width="200" />

Use [this guide](https://github.com/laurent22/joplin/issues/5691#issuecomment-1028241529) - basically downgrade to v2.5.5 and it works

**Comment by thurstylark** [comment]
This issue still affects me on v2.8.1 using my Pixel 2 XL running Android 11. Can this be reopened?

Here is the relevant portion of the debug log:

````
08-15T16:04:30,30,"RevisionService: ""maintenance: Done in 40889ms""
08-15T16:04:30,30,"RevisionService: ""collectRevisions: Created revisions for 0 notes""
08-15T16:04:30,30,"RevisionService: ""maintenance: Done in 27ms""
08-15T16:04:30,30,"RevisionService: ""maintenance: Service is enabled""
08-15T16:04:30,30,"RevisionService: ""maintenance: Starting...""
08-15T16:03:49,30,"RevisionService: ""maintenance: Service is enabled""
08-15T16:03:49,30,"RevisionService: ""maintenance: Starting...""
08-15T16:03:30,30,"RevisionService: ""maintenance: Done in 24068ms""
08-15T16:03:30,30,"RevisionService: ""collectRevisions: Created revisions for 0 notes""
08-15T16:03:06,30,"RevisionService: ""maintenance: Service is enabled""
08-15T16:03:06,30,"RevisionService: ""maintenance: Starting...""
08-15T16:02:13,30,"RevisionService: ""maintenance: Done in 66259ms""
08-15T16:02:13,30,"RevisionService: ""collectRevisions: Created revisions for 0 notes""
08-15T16:02:13,30,"RevisionService: ""maintenance: Done in 33ms""
08-15T16:02:13,30,"RevisionService: ""maintenance: Service is enabled""
08-15T16:02:13,30,"RevisionService: ""maintenance: Starting...""
08-15T16:01:07,30,"RevisionService: ""maintenance: Starting...""
08-15T16:01:07,30,"RevisionService: ""maintenance: Service is enabled""
08-15T16:00:09,30,"""DecryptionWorker: cannot start because no master key is currently loaded.""
08-15T16:00:08,20,"Synchronizer: ""There was some errors:""
08-15T16:00:08,20,"Synchronizer: ""Error: Error: Operation not permitted. Path: /storage/emulated/0/syncthing/joplin/locks/2_2_9fd0adbde1bb414bb7c553dbb0083b84.json
Code: EUNSPECIFIED
value@index.android.bundle:1108:275
index.android.bundle:1108:4056
p@index.android.bundle:83:588
index.android.bundle:83:1893
p@index.android.bundle:83:588
o@index.android.bundle:83:1069
index.android.bundle:83:1241
u@index.android.bundle:77:157
index.android.bundle:77:866
index.android.bundle:85:1672
k@index.android.bundle:85:500
w@index.android.bundle:85:898
callReactNativeMicrotasks@index.android.bundle:85:3054
value@index.android.bundle:38:2868
index.android.bundle:38:960
value@index.android.bundle:38:2504
value@index.android.bundle:38:919
value@[native code]
value@[native code]""
````

Full debug log attached

[syncReport-1660597473613.txt](https://github.com/laurent22/joplin/files/9339872/syncReport-1660597473613.txt)

**Comment by leder11011** [comment]
bump!

**Comment by OttovBiscuit** [comment]
Issue Still Present on Android 13 as of Jan 31st 2023
Device: Google Pixel 6 Pro
Joplin v.2.8.1
Database v41

**Comment by AlbertBaubleDeem** [comment]
Issue is still present on:
Android version
13 TKQ1.220829.002
Joplin 2.10.8
Database v42

As per communication in Issue #5779 the fix has been committed, but the version containing it hasn't yet been released.

As this issue is addressed in multiple threads, I am posting a link to the current workaround using adb for the sake of easier search:

https://github.com/laurent22/joplin/issues/5779#issuecomment-1078460620

It worked for me.
